mautic/mautic
# Support for company fields in tracking code pageview event

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | y
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | https://github.com/mautic/documentation/pull/266
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/5499
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )


[//]: # ( Required: )
#### Description:
Added support for company fields in     mt('send', 'pageview');
Inspirated by SubmissionModel https://github.com/mautic/mautic/blob/fed5d81bda03e25ca0d77108ca07b05de854c12a/app/bundles/FormBundle/Model/SubmissionModel.php#L1017

This PR also added more priority to companyname fields like take companyname from companyemail.

https://github.com/kuzmany/mautic/blob/1c1b1822aef2bec3f9ddebe0df8b7e0d7204569e/app/bundles/LeadBundle/Helper/IdentifyCompanyHelper.php#L76

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1.  Mark company fields as Publicly updatable  
2.  Use 
 `   mt('send', 'pageview', {companyemail: 'mycompany@mycompany.com', companywebsite: 'https://mautic.com/'});`

More in docs https://mautic.org/docs/en/contacts/contact_monitoring.html  (part Tracking of custom parameters)

3. See fields wasn't update in tracking contact

#### Steps to test this PR:
1.  Repeat alls teps
2.  See company fields was updated in Mautic
